### PR TITLE
add support for "additional-labels"

### PR DESCRIPTION
**Pull request by Rakshith-R**
add support for "additional-labels"
    
Signed-off-by: Niels de Vos <ndevos@ibm.com>
Signed-off-by: Rakshith R <rar@redhat.com>

We ran some tests at https://github.com/ceph/ceph-csi for the startsWith and if statement fix.

The new config "additional-labels" necessity was released since mergifyio runs only a subset of
rules on such merge queue draft prs.
Refer: https://github.com/ceph/ceph-csi/pull/3835#issuecomment-1549020177

/cc @sileht @jd @DouglasBlackwood

### CODE REVIEW

**Comment by Rakshith-R** [comment]
cc @nixpanic

File: action.yml [hunk 1]
```diff
@@ -5,12 +5,16 @@ inputs:
     description: 'Comma separated list of labels to copy (all if empty)'
     required: true
     default: ''
+  addional-labels:
```
Comment (sileht) [inline action.yml]:
```suggestion
  additional-labels:
```

File: action.yml [hunk 2]
```diff
@@ -20,6 +24,9 @@ runs:
           gh pr view --json labels -q '.labels[]|.name' ${REPOSITORY_URL}/pull/$pr_number | while read label ; do
             if [[ -z "$labels" ]] || [[ ",$labels," =~ ",$label," ]]; then
               gh pr edit --add-label "$label" $MERGE_QUEUE_PR_URL
-            if
+            fi
+          done
+          for additional-label in $(echo $additional-labels | sed "s/,/ /g") ; do
```
Comment (sileht) [inline action.yml]: Can we set the env with`ADDITIONAL_LABELS: ${{ additional-labels }}` and:

```suggestion
          for additional-label in $(echo $ADDITIONAL_LABELS | sed "s/,/ /g") ; do
```

File: action.yml [hunk 3]
```diff
@@ -20,6 +24,9 @@ runs:
           gh pr view --json labels -q '.labels[]|.name' ${REPOSITORY_URL}/pull/$pr_number | while read label ; do
             if [[ -z "$labels" ]] || [[ ",$labels," =~ ",$label," ]]; then
               gh pr edit --add-label "$label" $MERGE_QUEUE_PR_URL
-            if
+            fi
```
Comment (sileht) [inline action.yml]: This should be in separate pull request.

File: action.yml [hunk 4]
```diff
@@ -5,12 +5,16 @@ inputs:
...
+    description: 'Comma separated list of additional labels to add to the merge queue PR'
+    required: false
+    default: ''
 runs:
   using: "composite"
   steps:
     - name: Copying labels
       shell: bash
-      if: startsWith('mergify/merge-queue/', github.head_ref)
+      if: startsWith(github.head_ref, 'mergify/merge-queue/')
```
Comment (sileht) [inline action.yml]: This should be in separate pull request.

**Review by sileht — COMMENTED** [review]
Thanks for the contribution! I make a couple of remarks.

**Review by sileht — COMMENTED** [review]
Thanks for the contribution! I make a couple of remarks.

**Comment by Rakshith-R** [comment]
> Thanks for the contribution! I make a couple of remarks.

:+1: , updated this pr and opened two more separate prs.

**Review by sileht — APPROVED** [review]
Thanks!
